apps/src/dance/ai/DanceAiModalHeader.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {useMemo} from 'react';
+import moduleStyles from './dance-ai-modal.module.scss';
+import aiBotMini from '@cdo/static/dance/ai/bot/ai-bot-mini.svg';
+const i18n = require('../locale');
+
+import {DanceAiModelItem} from './types';
+
+import EmojiIcon from './EmojiIcon';
+
+interface DanceAiModalHeaderProps {
+  onClose: () => void;
+  handleStartOverClick: () => void;
+  selectedEmojis: DanceAiModelItem[];
+  slotCount: number;
+}
+
+const DanceAiModalHeader: React.FunctionComponent<DanceAiModalHeaderProps> = ({
+  onClose,
+  handleStartOverClick,
+  selectedEmojis,
+  slotCount,
+}) => {
+  const headerValue = useMemo(() => {
+    const filledSelectedEmojis: (DanceAiModelItem | undefined)[] = [
+      ...selectedEmojis,
+    ];
+    filledSelectedEmojis.length = slotCount;
+    filledSelectedEmojis.fill(undefined, selectedEmojis.length, slotCount);
+
+    return (
+      <div
+        className={moduleStyles.inputsContainer}
+        tabIndex={0}
+        onClick={handleStartOverClick}
+      >
+        {filledSelectedEmojis.map((item, index) => (
+          <div key={item?.id || index} className={moduleStyles.emojiSlot}>
+            {item && (
+              <EmojiIcon item={item} className={moduleStyles.emojiSlotIcon} />
+            )}
+          </div>
+        ))}
+      </div>
+    );
+  }, [handleStartOverClick, selectedEmojis, slotCount]);
```

Comment:
This was originally a function that was only called once. That would cause e-renders, so was better as a `useCallback`. But since it was only called once, was better as a `useMemo`

Should also probably be combined with `headerContent` down below into a single `useMemo` instead.